app/services/parent_profile_associator.rb

```diff
@@ -6,26 +6,69 @@ class << self
     def run!
       Profile.transaction do
         Profile.where.not(account: nil).find_each do |profile|
-          profile.update!(parent_profile: find_parent(profile))
+          parent = find_parent(profile)
+          next if profile_already_in_account!(profile, parent)
+
+          profile.update!(parent_profile: parent, benchmark: parent.benchmark)
         end
       end
     end
 
     private
 
+    def profile_already_in_account!(profile, parent)
+      duplicate_profiles = profile.account.profiles.where(
+        ref_id: profile.ref_id,
+        benchmark_id: parent.benchmark.id
+      ).count
+      return false unless duplicate_profiles > 1
+
+      profile.delete_all_test_results = true
+      profile.destroy
+    end
+
     def find_parent(profile)
-      parents = Profile.where(ref_id: profile.ref_id,
-                              benchmark_id: profile.benchmark_id,
-                              account_id: nil)
-      raise not_found(profile) unless parents.one?
+      logger.info("Finding parents for profile #{profile.id}")
+      parents = find_in_same_benchmark(profile)
+      parents = find_in_other_benchmarks(profile) if parents.blank?
+
+      raise not_found(profile) unless parents.any?
```

Comment:
So I have discovered some bad data in the phony benchmark; We have 51 profiles in the prod dump that have an incorrect description - it's the benchmark description rather than the profile description. There are no other profiles outside the phony benchmark with this problem, so it seems we've fixed the bug in production already; however, we need to handle these profiles.

Here's an example profile from the prod dump with this problem:

```rb
=> #<Profile:0x000055872542f200
 id: "0096f7cd-aebb-49a6-a91e-8caba583800b",
 name: "OSPP - Protection Profile for General Purpose Operating Systems v. 4.2",
 ref_id: "xccdf_org.ssgproject.content_profile_ospp42",
 created_at: Thu, 09 May 2019 21:21:45 UTC +00:00,
 updated_at: Fri, 04 Oct 2019 08:32:09 UTC +00:00,
 description:
  "This guide presents a catalog of security-relevant\nconfiguration settings for Red Hat Enterprise Linux 7. It is a rendering of\ncontent structured in the eXtensible Configuration Checklist Description Format (XCCDF)\nin order to support security automation.  The SCAP content is\nis available in the scap-security-guide package which is developed at\n\n    https://www.open-scap.org/security-policies/scap-security-guide.\n\nProviding system administrators with such guidance informs them how to securely\nconfigure systems under their control in a variety of network roles. Policy\nmakers and baseline creators can use this catalog of settings, with its\nassociated references to higher-level security control catalogs, in order to\nassist them in security baseline creation. This guide is a catalog, not a\nchecklist, and satisfaction of every item is not likely to be possible or\nsensible in many operational scenarios. However, the XCCDF format enables\ngranular selection and adjustment of settings, and their association with OVAL\nand OCIL content provides an automated checking capability. Transformations of\nthis document, and its associated automated checking content, are capable of\nproviding baselines that meet a diverse set of policy objectives. Some example\nXCCDF Profiles, which are selections of items that form checklists and\ncan be used as baselines, are available with this guide. They can be\nprocessed, in an automated fashion, with tools that support the Security\nContent Automation Protocol (SCAP). The DISA STIG for Red Hat Enterprise Linux 7,\nwhich provides required settings for US Department of Defense systems, is\none example of a baseline created from this guidance.\n",
 account_id: "9b22a83d-57af-4b67-9430-148fc4bdc251",
 compliance_threshold: 100.0,
 business_objective_id: nil,
 benchmark_id: "fe07c3bc-0f58-47ac-bdf6-e2b137ac5677">
```

As you can see, this description is for the benchmark, not the profile. Of the 51 profiles with this issue, 3 are hipaa and thus their benchmark cannot be determined. We need to decide to either associate them to a random benchmark, ignore them, or delete them. Those are the 3 options I see as a way forward. If we keep this `raise`, we won't be able to finish running this migration.

Reply:
This could be one solution (tested and seems to work fine):

```diff
diff --git a/app/services/parent_profile_associator.rb b/app/services/parent_profile_associator.rb
index 76e9ca0..0e00844 100644
--- a/app/services/parent_profile_associator.rb
+++ b/app/services/parent_profile_associator.rb
@@ -7,6 +7,7 @@ class ParentProfileAssociator
       Profile.transaction do
         Profile.where.not(account: nil).find_each do |profile|
           parent = find_parent(profile)
+          next unless parent
           next if profile_already_in_account!(profile, parent)
 
           profile.update!(parent_profile: parent, benchmark: parent.benchmark)
@@ -32,7 +33,7 @@ class ParentProfileAssociator
       parents = find_in_same_benchmark(profile)
       parents = find_in_other_benchmarks(profile) if parents.blank?
 
-      raise not_found(profile) unless parents.any?
+      return unless parents.any?
 
       parent = find_most_recent_parent(parents)
       logger.info("Found parent #{parent.id} for profile #{profile.id}")
```